reason-parser/src/reason_pprint_ast.ml

```diff
@@ -2566,12 +2566,12 @@ class printer  ()= object(self:'self)
                 first::(List.map (formatOneTypeDefStandard (atom "and")) (tlhd::tltl))
               )
 
-  method type_variant_leaf ?opt_ampersand:(a=false) ?polymorphic:(p=false) = self#type_variant_leaf1 a p true
-  method type_variant_leaf_nobar ?opt_ampersand:(a=false) ?polymorphic:(p=false) = self#type_variant_leaf1 a p false
+  method type_variant_leaf ?attrs_on_leaf:(attrs=[]) ?opt_ampersand:(a=false) ?polymorphic:(p=false) = self#type_variant_leaf1 attrs a p true
+  method type_variant_leaf_nobar ?opt_ampersand:(a=false) ?polymorphic:(p=false) = self#type_variant_leaf1 [] a p false
 
   (* TODOATTRIBUTES: Attributes on the entire variant leaf are likely
    * not parsed or printed correctly. *)
-  method type_variant_leaf1 opt_ampersand polymorphic print_bar x =
+  method type_variant_leaf1 attrs_on_leaf opt_ampersand polymorphic print_bar x =
```

Comment:
I haven't looked at this code in a while, so my memory is rusty here, but how is it that we have attributes that are on the "entire leaf", joined with the "standard attributes", which are all printed as equals, and then later when we parse we somehow can recover which was which?